apps/ewallet_config/lib/ewallet_config/config.ex

```diff
@@ -102,9 +102,28 @@ defmodule EWalletConfig.Config do
 
   @spec update(Map.t(), Atom.t()) :: [{:ok, %Setting{}} | {:error, Atom.t()}]
   def update(attrs, pid \\ __MODULE__) do
-    GenServer.call(pid, {:update_and_reload, attrs})
+    {config_pid, attrs} = get_config_pid(attrs)
+
+    case Mix.env() == :test do
+      true ->
+        GenServer.call(config_pid || pid, {:update_and_reload, attrs})
+
+      false ->
+        GenServer.call(__MODULE__, {:update_and_reload, attrs})
+    end
   end
 
+  defp get_config_pid(%{config_pid: config_pid} = attrs),
+    do: {config_pid, Map.delete(attrs, :config_pid)}
+
+  defp get_config_pid(%{"config_pid" => config_pid} = attrs),
+    do: {config_pid, Map.delete(attrs, "config_pid")}
+
+  defp get_config_pid([{:config_pid, config_pid}] = attrs),
```

Comment:
I think this works only if the list is exactly `[{:config_pid, config_pid}]`. It won't work with `[{:some_other_key, some_other_value}, {:config_pid, config_pid}]`, etc.

Probably need to use `Keyword.fetch/3` or `Keyword.get/3` within the function implementation instead.